### DISCUSSION TITLE
Moonlight

### DISCUSSION THREAD

**Discussion by krahabb** [krahabb/meross_lan#375] [Releases]
Moonlight

This is a huge release  in terms of code redesign bringing in a lot of features for diagnostic/maintenance of devices together with state-of-the-art bug fixes and some new features/supported devices. Let's see what's new:

# New features
- supported Meross multi-factor-authentication (for accounts so configured) (#372)
- supported the mts960 Smart Thermostat plug (#356)
- supported the map100 Smart Air Purifier (#388)
- supported the Child-Lock feature for devices exposing it
- added the ability to configure device MQTT binding (so to link it to the same HA MQTT broker and 'discard' Meross cloud MQTT) without the need of external utilities/app

# Improvements
- device communication has been improved in order to reduce the number of exchanged messages thus improving latency and (hopefully) overall communication performance
- added device timezone check and configuration to ease solve some (mss310 or any metering plug) issues (#367)
- logging and tracing completely re-done to give better control (at the device configuration level)
- added 'diagnostic sensors' (configurable per device) to automatically extract and show in HA UI the data exposed by any device for (still) unsupported features

# Fixes
- almost any pending issue has been fixed (check its state in the [issues](https://github.com/krahabb/meross_lan/issues) registry)

**Comment by MattCheetham** [comment]
Fixed the issues for my MSS315 plugs relating to timezone. Now able to add them to home via Matter and use this to show the data without connecting them to cloud.
